app/Http/Controllers/UserController.php

```diff
@@ -42,8 +42,10 @@ public function show(User $user): View
     public function update(UserUpdateRequest $request): RedirectResponse
     {
         $user = type($request->user())->as(User::class);
+        $data = $request->validated();
+        $data['prefers_questions'] = $request->has('prefers_questions');
 
-        $user->fill($request->validated());
```

Comment:
Since the `prefers_questions` is validated as a boolean. Why not preserve the original code?

Reply:
Because if the checkbox is unchecked the form doesn't include it in the payload. So I check if the prefers_questions field is set it will store the value, If not it will store 0 or false.

Reply:
You can use radio or select if you want to preserve previous code

Reply:
Okay, I think I will use select.